src/main/java/dev/dokan/dokan_java/examples/DirListingFileSystem.java

```diff
@@ -242,7 +241,9 @@ public int getVolumeInformation(Pointer rawVolumeNameBuffer, int rawVolumeNameSi
 
     private Path getrootedPath(WString rawPath) {
         String unixPath = rawPath.toString().replace('\\', '/');
-        String relativeUnixPath = CharMatcher.is('/').trimLeadingFrom(unixPath);
+        String relativeUnixPath = unixPath.startsWith("/")?
+                ("/".equals(unixPath)?"":unixPath.substring(1))
+                :unixPath;
```

Comment:
Nesting statements with the elvis operator is confusing. I suggest to split it up even if it will result in more code lines. Maybe something like

```Java
String relativeUnixPath = unixPath;
if(unixPath.startsWith("/")){ 
        relativeUnixPath =  unixPath.length()==1?"":unixPath.substring(1); // if it is already the root, we return the empty string
}
```

Reply:
done